src/controller/mod.rs

```diff
@@ -80,14 +68,34 @@ impl Controller {
                 context.set_sidebar_stage(SidebarStage::PROJECTS);
                 State::Continue
             }
+            Key::Up => {
```

Comment:
The controller should not affect the sidebar when it is in the main stage. This requirement is not satisfied currently. Let's fix this.

Reply:
Done.